sktime/utils/_testing/deep_equals.py

```diff
@@ -108,6 +119,9 @@ def ret(is_equal, msg):
             return ret(False, f".dtype, x.dtype = {x.dtype} != y.dtype = {y.dtype}")
         return ret(np.array_equal(x, y, equal_nan=True), ".values")
     # recursion through lists, tuples and dicts
+    elif check_xarray(x, y)[0]:
```

Comment:
Do you have an idea how to solve this in a more elegant way? 

I need to wrap this check behind a method for checking if xarray is installed.

Reply:
* the last condition defaults to `__eq__`, which is called on the object. Does that not suffice, or does that check identity in reference rather than just in value?
* otherwise no good idea, but I would really like the `deep_equals` module decoupled from any soft dependencies (not just conditionally decoupled but absolutely)

I assume this is because tests are using this, and it breaks without the modification?

"best practice" would be using a dispatch pattern, but that would be a major refactor.
Could do "cheap" dictionary dispatch, on type, like in datatypes, or "full blown" dispatch.

Reply:
* `__eq__` does element wise comparison as `np.arrays`. Consequently, the elif fails since the result is ambiguous. An alternative would be wrapping the `__eq__` with `np.all` or `np.any`. However, this only checks if the values are the same and do not perform checks on the index or the columns, which I think is expected when a deep_equals is called.

Yes, the tests break without this modification.

Reply:
I tested it, we can remove this check_xarray and replacing the last elif statement with `np.any(x != y)`. It would enable to check the equality of the values of xr.DataArrays and if `x!=y` results into a bool, it still works since `np.any(True)` is `True` and `np.any(False)` is `False`.